DataManagementSystem/DB/FileCatalogComponents/MultiVODirectoryMetadata.py

```diff
@@ -0,0 +1,39 @@
+""" DIRAC Multi VO FileCatalog plugin class to manage file metadata for multiple VO.
+"""
+
+__RCSID__ = "$Id$"
+
+from DIRAC.DataManagementSystem.DB.FileCatalogComponents.DirectoryMetadata import DirectoryMetadata
+from DIRAC.ConfigurationSystem.Client.Helpers import Registry
+
+
+class MultiVODirectoryMetadata(DirectoryMetadata):
+  """
+  MULti-VO FileCatalog plugin implementation.
+  """
+
+  def __init__(self, database=None):
+    DirectoryMetadata.__init__(self, database=database)
+
+  def _getMetaName(self, meta, credDict):
```

Comment:
Maybe this and getMetaNameSuffix can be turned into a Mixin class and added to this and MultiVOFileMetadata

Reply:
Yes, could do that. We would need 2 mixing classes: BaseMixin (pass through) and DerivedMixin one to maintain the suffix adding and removal. Base FileMetadata and DirectoryMetadata classes would both inherit from the BaseMixin and the Multi VO counterparts would inherit from the DerivedMixin. If it is what you meant I can implement this. I'll obviously find suitable names for the mixin classes.

Reply:
Yes. Reduces code duplication